### DISCUSSION TITLE
Authenticate via GitHub?

### DISCUSSION THREAD

**Discussion by FGasper** [community/community#124692] [npm]
Hello,

I recently tried to log in to NPM. I did a password reset, and along the way I noticed that GitHub now seems to control NPM.

That being the case, would it be possible to authenticate to NPM via GitHub?

Thanks!

**Comment by dom557** [comment]
Yes, it's possible to authenticate to npm via GitHub. npm supports authentication using a GitHub personal access token. You can create a personal access token in your GitHub account settings and then use it to authenticate npm by running:

```
npm login --registry=https://npm.pkg.github.com --scope=@OWNER
```

Replace `@OWNER` with your GitHub username or the organization name if you're publishing packages under an organization. It will prompt you to enter your GitHub username, password, and personal access token. Once authenticated, npm will store your credentials securely and allow you to publish and install packages from GitHub Packages registry.

**Reply by FGasper** [reply]
Ah ok.

What about, though, authenticating on the website via OAuth? I can log in to dev.to and other sites via GitHub .. it'd be great for NPM to expose the same functionality.

Thanks!

**Reply by dom557** [reply]
I think you  can authenticate to npm via GitHub on the npm website using OAuth. To do this, go to [npmjs.com](https://www.npmjs.com/) and click on the "Sign In" button in the top right corner. On the sign-in page, select the "Sign in with GitHub" option. This will redirect you to GitHub, where you will need to log in (if not already logged in) and authorize npm to access your GitHub account. After authorization, you will be redirected back to the npm website, where you should be logged in. This method leverages OAuth to link your GitHub account to your npm account, allowing you to use your GitHub credentials to access npm.

**Reply by FGasper** [reply]
@dom557 I don’t see a “Sign in with GitHub” option:
<img width="937" alt="image" src="https://github.com/community/community/assets/844867/e83a02af-3d3a-4513-a891-31609c62a7fe">

**Reply by dom557** [reply]
@FGasper If you don't see a "Sign in with GitHub" option on the npm website, it likely means that npm hasn't integrated this feature for web-based authentication yet. However, you can still authenticate using your GitHub credentials via the command line. First, create a GitHub Personal Access Token by going to your GitHub account settings, navigating to "Developer settings" > "Personal access tokens," and generating a new token with the necessary permissions for npm. Then, open your terminal and run `npm login --registry=https://npm.pkg.github.com --scope=@OWNER`, replacing `@OWNER` with your GitHub username or the organization name. You will be prompted to enter your GitHub username, password, and the personal access token you created. This method allows npm to securely store your credentials, enabling you to publish and install packages from the GitHub Packages registry. If npm adds the "Sign in with GitHub" option on their website in the future, it will make the process even more straightforward.

**Reply by alexbuznik-stark** [reply]
I only receive the username and password prompt, and after I enter username - it skips password and has "loading" animation, but nothing happens

**Comment by FGasper** [comment]
> it likely means that npm hasn't integrated this feature for web-based authentication yet

That makes sense, yeah. Is this the proper forum to request that feature? If not, where should I direct that request?

It sounds like they have the “plumbing” in place for this, so hopefully it’s not too tall of an ask. :)

**Reply by dom557** [reply]
@FGasper It's great to hear that makes sense :) ,  This forum might not be the most effective place to request new features. Instead, you can direct your request to the npm or GitHub support teams or submit it as a feature request on their respective repositories. 

For npm, you can visit their [support page](https://www.npmjs.com/support) to submit a request or feedback. Additionally, you can open an issue on the [npm GitHub repository](https://github.com/npm/cli/issues) to suggest new features. 

For GitHub-related features, you can submit feedback through the [GitHub Support page](https://support.github.com/contact/feedback) or open an issue on the [GitHub feedback repository](https://github.com/github/feedback/discussions).

Given that the underlying OAuth mechanisms are already in place, implementing this feature might not be a significant task for their development teams. Good luck, and I hope your feature request gets implemented soon!

**Reply by dom557** [reply]
@FGasper  If you use Twitter, you can also tweet your suggestion to the npm team (@npmjs) , good luck .

**Reply by FGasper** [reply]
@dom557 The support page says that suggestions to improve NPM should actually go to this form:
<img width="381" alt="image" src="https://github.com/community/community/assets/844867/05f0b7ff-c077-4c57-829c-cb4705e4ad92">
The GitHub repo you referenced seems to be the `npm` CLI tool. But you’re saying that that’s also a proper forum to suggest changes to the website?

**Reply by dom557** [reply]
@FGasper Thanks for pointing that out. If the support page directs you to a specific form for suggestions, that should be your primary channel. Submit your request through the form at [npm Suggestion Form](https://www.npmjs.com/support) to ensure it reaches the right team. Additionally, you can open an issue on the [npm CLI GitHub repository](https://github.com/npm/cli/issues) for broader feedback, including website changes. Just make sure to specify that your request pertains to the website. This dual approach helps ensure your suggestion is seen by the right people.

**Reply by dom557** [reply]
@FGasper the [npm CLI GitHub repository](https://github.com/npm/cli/issues) should look like this :
![image](https://github.com/community/community/assets/124943524/9ce5dab4-f816-48da-8cb0-c4e4452564cb)
